R/Figures.R

```diff
@@ -1138,6 +1138,385 @@ TADA_TwoCharacteristicScatterplot <- function(.data, id_cols = "TADA.ComparableD
+#' # If no groups are selected, return the 4 groups (by MonitoringLocationName) with the greatest number of field count results for that TADA.ComparableDataIdentifier
+#' TADA_MultiScatterplot(df, id_cols = c("TADA.ComparableDataIdentifier", "MonitoringLocationName"))
+#' # Both commands below will return a single scatterplot for just the single comparable data identifier. 
+#' TADA_MultiScatterplot(df, id_cols = c("TADA.ComparableDataIdentifier", "TADA.ComparableDataIdentifier"))
+#' TADA_MultiScatterplot(df)
+#' 
+
+TADA_MultiScatterplot <- function(.data, id_cols = c("TADA.ComparableDataIdentifier"), groups = NULL) {
+  # check .data is data.frame
+  TADA_CheckType(.data, "data.frame", "Input object")
+  
+  # check .data has required columns
+  TADA_CheckColumns(.data, id_cols)
+  
+  # check .data has required columns
+  reqcols <- c(
+    "TADA.ResultMeasureValue",
+    "TADA.ResultMeasure.MeasureUnitCode",
+    "ActivityStartDate"
+  )
+  
+  # check .data has required columns
+  TADA_CheckColumns(.data, reqcols)
+  
+  # if left blank, ensure comparable data identifier is in the id_cols vector
+  if (is.null(id_cols) | length(unique(id_cols)) == 1) {
+    id_cols <- c("TADA.ComparableDataIdentifier","TADA.ComparableDataIdentifier")
+  }
+  id_cols <- c("TADA.ComparableDataIdentifier", vecsets::vsetdiff(id_cols, "TADA.ComparableDataIdentifier"))
+  
+  # if id_cols are specified by user and the comparable data identifier is not selected.
+  if (!"TADA.ComparableDataIdentifier" %in% id_cols) {
+    stop("Note: TADA.ComparableDataIdentifier not found in id_cols argument.")
+  }
+  
+  # Only allows for 2 column selections in the id_cols vector, with TADA.ComparableDataIdentifier being a required value
+  if (length(id_cols) > 2) {
+    stop("id_cols argument can only be a maximum of length two.")
+  }
+  
+  # Runs if more than 1 Tada.ComparableDataIdentifiers are found in a dataframe. Prompts for user selection of a  single characteristic in the dataframe
+  # var <- NULL
+  # id2 <- NULL
+  # if(length(unlist(unique(.data[, id_cols[1]]))) > 1){
+  #   var <- utils::menu(unlist(unique(.data[, id_cols[1]])), title = "Please specify Characteristic to plot: ")
+  #   assign("id2", unlist(unique(.data[, id_cols[1]]))[var])
+  # }
+  
+  # if groups are not specified, select the top 4 groups to plot.
+  if (length(id_cols) == 2 & is.null(groups)) {
+    assign("groups", names((utils::head(sort(table(.data[,id_cols[2]]), decreasing = TRUE), 4))))
+    print("Note: No 'groups' specified. Plotting up to top 4 'groups' based on the number of field counts")
+  }
+  
+  # check that groups are in id_cols
+  id <- unlist(unique(.data[, id_cols[2]]))
+  if (any(!groups %in% id)) {
+    stop("The 'groups' vector contains one or more inputs that are not found within your input dataset. Check spelling and try again.")
```

Comment:
Would it make sense to make this "stop" message more specific by pasting the strings that were not found in the input data set in to the message?

Usually for "stop" type messages (and in many other cases, although I don't know that as a team we've had a specific discussion about this convention yet), we print the name of the function in the message to help users know where thing went wrong. Example: "TADA_MultiScatterplot: The 'groups' vector...."